nilearn/plotting/cm.py

```diff
@@ -54,40 +54,36 @@ def mix_colormaps(fg, bg):
 
 def _rotate_cmap(cmap, swap_order=("green", "red", "blue")):
     """Swap the colors of a colormap."""
-    orig_cdict = cmap._segmentdata.copy()
+    cdict = cmap._segmentdata.copy()
 
-    cdict = {}
-    cdict["green"] = list(orig_cdict[swap_order[0]])
-    cdict["blue"] = list(orig_cdict[swap_order[1]])
-    cdict["red"] = list(orig_cdict[swap_order[2]])
-
-    return cdict
+    return {
+        "green": list(cdict[swap_order[0]]),
+        "blue": list(cdict[swap_order[1]]),
+        "red": list(cdict[swap_order[2]]),
+    }
 
 
 def _pigtailed_cmap(cmap, swap_order=("green", "red", "blue")):
     """Make a new colormap by concatenating a colormap with its reverse."""
-    orig_cdict = cmap._segmentdata.copy()
-
-    cdict = {}
-    cdict["green"] = [
-        (0.5 * (1 - p), c1, c2)
-        for (p, c1, c2) in reversed(orig_cdict[swap_order[0]])
-    ]
-    cdict["blue"] = [
-        (0.5 * (1 - p), c1, c2)
-        for (p, c1, c2) in reversed(orig_cdict[swap_order[1]])
-    ]
-    cdict["red"] = [
-        (0.5 * (1 - p), c1, c2)
-        for (p, c1, c2) in reversed(orig_cdict[swap_order[2]])
-    ]
-
-    for color in ("red", "green", "blue"):
-        cdict[color].extend(
-            [(0.5 * (1 + p), c1, c2) for (p, c1, c2) in orig_cdict[color]]
-        )
+    cdict = cmap._segmentdata.copy()
 
-    return cdict
+    return {
+        "green": [
+            (0.5 * (1 - p), c1, c2)
+            for (p, c1, c2) in reversed(cdict[swap_order[0]])
+        ]
+        + [(0.5 * (1 + p), c1, c2) for (p, c1, c2) in cdict["green"]],
```

Comment:
do you think you could extract this into a small function with explicit name and variable names that are not just one letter because at the moment it is pretty hard to understand what it going on?

Reply:
I have created a function but am not sure about meaningful names. Indeed I haven't been able to understand enough of the code yet and derive names that would be standard or meaningful 😀

Reply:
haha: that makes me feel so less stupid !!
thanks